packages/aws-cdk/test/integ/cli/test-persistent-stack.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -euo pipefail
+scriptdir=$(cd $(dirname $0) && pwd)
+source ${scriptdir}/common.bash
+# ----------------------------------------------------------
+
+setup
+
+echo "Creating/updating the persistent stack"
+cdk deploy ${STACK_NAME_PREFIX}-PersistentStack
+
+# NOTE: No 'destroy' on that stack on purpose
+echo "✅  success"
```

Comment:
Im just wondering whether we should add some assertions here on the actual resource properties. The command itslef might succeed but have an undesired affect.

For exmaple, what happens if we suddenly decide to change the default VPC name? Should we be changing existing VPC names as well? I guess this relates to the bigger engineering effort you mentioned. 

I'm also ok with just saying that this test simply verifies the command execution, and ignores any "behvaior changes" that might happen. But I wonder if its worth adding these sepcific assertions now and build on this pattern as we continue to add these compatibilty tests.

This also brings up the question of how we treat implicit updates in general, ones that the user didn't specifically request - do we make those?

I might be out of base with the example I gave since it feels a bit extreme, but I imagine you see where i'm going with this.

Reply:
This test is specifically intended to make sure that stacks don't become undeployable, which happens to have happened in the case we ran into. Verifying properties about the stack may be correct, but it may also introduce annoying churn (we want to make a legitimate change and will be kneecapped at some later point by failing tests).

In any case, given that we use desired-state configuration, I would say that validating properties of the "new" situation is the responsibility of the construct library (so unit tests and integration tests *there*).

> This also brings up the question of how we treat implicit updates in general

It is kinda the value proposition of the CDK that you can get those. It's our responsibility to make sure the updates you get in that way won't suck for the customer.

In the ideal case scenario, we'd update the space of *all possible* applications in-place when a new CDK version comes out, and verify that:

* No data loss occurred
* There was no interruption of service

Since testing that is obviously intractable, we're going to have to scale it down a little. I have some inkling of what that *might* look like (potentially some kind of reference application with canaries which we update in place on every new CDK release and see the canaries can still read/write data). None of that is small potatoes work, and it doesn't generalize well.

I wonder if we can do something clever, but I'm afraid it just going to be a lot of construct-specific grunt work.

Reply:
Yeap, makes a lot of sense 👍 

Regarding the in-place updates, I hope we can rely on [CFN Update Behavior](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html). 

Basically not doing any implicit updates that dont fall under the *"Update with No Interruption"* behavior. If such updates are required, we should probably be prompting the user for approval.

But all that is for a different time :)